ocaml/xcp-rrdd/scripts/rrdd/rrdd.py

```diff
@@ -44,267 +44,288 @@
+
+    def next_reading(self):
+        """Return the time period after which the next reading for this plugin
+        will take place."""
+        return self.dispatcher.next_reading({"uid": self.uid})
+
+    def wait_until_next_reading(self, neg_shift=1):
+        """The xcp-rrdd daemon reads the files written by registered plugins in
+        pre-determined time intervals. This function coordinates this timing with
+        the daemon, and wakes up just before the next such reading occurs. This
+        way, the plugin can provide freshly collected data. The neg_shift argument
+        specifies (in seconds; can be a fraction; default is 1) how long before
+        the next reading this function return. Note that it is up to the plugin
+        author to choose a value for neg_shift that is at least as large the time
+        it takes for the plugin to collect its data; however, it should also not
+        be much larger, since this decreases the freshness of the data."""
+        while True:
+            try:
+                self.lazy_complete_init()
+                next_reading = self.register()
+                wait_time = next_reading - neg_shift
+                if wait_time < 0: wait_time %= self.frequency_in_seconds
+                time.sleep(wait_time)
+                return
+            except socket.error:
+                msg = "Failed to contact xcp-rrdd. Sleeping for 5 seconds .."
+                print >> sys.stderr, msg
+                time.sleep(5)
+
+    def update(self):
+        """Write all datasources specified (via set_datasource) since the last
+        call to this function. The datasources are written together with the
+        relevant metadata into the file agreed with rrdd."""
+        timestamp = int(time.time())
+        data_values = []
+        combined = dict()
+        data_checksum = crc32(pack(">Q", timestamp)) & 0xffffffff
+
+        for ds in sorted(self.datasources, key=lambda source: source.name):
+            value = self.pack_data(ds)
+            data_values.append(value)
+            data_checksum = crc32(value, data_checksum) & 0xffffffff
+            combined.update(ds.metadata)
+
+        metadata = {"datasources": combined}
+        metadata_json = json.dumps(metadata, sort_keys=True)
+        metadata_checksum = crc32(metadata_json) & 0xffffffff
+
+        self.dest.seek(0)
+        self.dest.write('DATASOURCES'.encode())
+        self.dest.write(pack(">LLLQ",
+                             data_checksum,
+                             metadata_checksum,
+                             len(self.datasources),
+                             timestamp))
+        for val in data_values:
+            # This is already big endian encoded
+            self.dest.write(val)
+
+        self.dest.write(pack(">L", len(metadata_json)))
+        self.dest.write(metadata_json.encode())
+        self.dest.flush()
+        self.datasources = []
+        time.sleep(
```

Comment:
why the linebreak here?

Reply:
I guess the reformat (which was automated) split the line.